### PR TITLE
fix: nan loss values 

### PR DESCRIPTION
**Pull request by srsawant34**
This resolves the issue #141. 

With previous code:
```
from losses import SupConLoss
import torch
import torch.nn.functional as F

# define loss with a temperature `temp`
temp = 0.8
criterion = SupConLoss(temperature=temp)

# features: [bsz, n_views, f_dim]
# `n_views` is the number of crops from each image
# better be L2 normalized in f_dim dimension
features = torch.randn(4,1,3)
features = F.normalize(features, p=2, dim=1)
# labels: [bsz]
labels = torch.tensor([0,1,1,3], dtype=float)

# SupContrast
loss = criterion(features, labels)
```
Outputs (before mean):
```
tensor([    nan, 36.9528, 36.9528,     nan])
```

With new changes:
```
from losses import SupConLoss
import torch
import torch.nn.functional as F

# define loss with a temperature `temp`
temp = 0.8
criterion = SupConLoss(temperature=temp)

# features: [bsz, n_views, f_dim]
# `n_views` is the number of crops from each image
# better be L2 normalized in f_dim dimension
features = torch.randn(4,1,3)
features = F.normalize(features, p=2, dim=1)
# labels: [bsz]
labels = torch.tensor([0,1,1,3], dtype=float)

# SupContrast
loss = criterion(features, labels)
```
Outputs (before mean):
```
tensor([-0.0000, 65.1030, 36.9528, -0.0000])
```

### CODE REVIEW

**Comment by srsawant34** [comment]
@HobbitLong Kindly look into this pull request. Thanks :)

**Comment by HobbitLong** [comment]
Hi, @srsawant34 , thanks for this commit.

Why the loss for the second item jumps from `36.9528` to `65.1030`?

**Comment by srsawant34** [comment]
@HobbitLong Kindly look into this, I would appreciate your time. Thanks

**Comment by HobbitLong** [comment]
@srsawant34, thank you for the contribution. Merged!
